engine/src/main/java/org/hibernate/search/analyzer/impl/AnalyzerReference.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.analyzer.impl;
+
+import org.apache.lucene.analysis.Analyzer;
+
+/**
+ * Reference an analayzer implementation.
+ * <p>
+ * It can be a reference to an {@link Analyzer} or a {@link RemoteAnalyzer}.
+ *
+ * @author Davide D'Alto
+ */
+public class AnalyzerReference {
```

Comment:
Ok, what I'd find more important is pushing instantiation of specific
reference types to the chosen backend instead of just using the provider
interface as some marker and acting based on that in core.